addons/mail/static/tests/discuss/core/new_message_separator.test.js

```diff
@@ -0,0 +1,222 @@
+    await click(".o-mail-DiscussSidebar-item", { text: "test" });
+    await contains(".o-mail-Discuss-threadName", { value: "test" });
+    await contains(".o-mail-Thread-newMessage hr + span", { count: 0, text: "New messages" });
+});
+
+test("show after receiving message out of odoo focus", async () => {
+    mockService("presence", () => ({
+        ...presenceService.start(),
+        isOdooFocused: () => false,
+    }));
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({ name: "Foreigner partner" });
+    const userId = pyEnv["res.users"].create({
+        name: "Foreigner user",
+        partner_id: partnerId,
+    });
+    const channelId = pyEnv["discuss.channel"].create({
+        channel_member_ids: [
+            Command.create({ message_unread_counter: 0, partner_id: serverState.partnerId }),
+            Command.create({ partner_id: partnerId }),
+        ],
+        channel_type: "channel",
+        name: "General",
+    });
+    const env = await start();
+    rpc = rpcWithEnv(env);
+    await openDiscuss(channelId);
+    await contains(".o-mail-Thread");
+    await contains(".o-mail-Thread-newMessage hr + span", { count: 0, text: "New messages" });
+    // simulate receiving a message
+    await withUser(userId, () =>
+        rpc("/mail/message/post", {
+            post_data: { body: "hu", message_type: "comment", subtype_xmlid: "mail.mt_comment" },
+            thread_id: channelId,
+            thread_model: "discuss.channel",
+        })
+    );
+    await contains(".o-mail-Message", { text: "hu" });
+    await contains(".o-mail-Thread-newMessage hr + span", { text: "New messages" });
+    await contains(".o-mail-Thread-newMessage ~ .o-mail-Message", { text: "hu" });
+});
+
+test("remove when user sends a message", async () => {
+    const pyEnv = await startServer();
+    const bobPartnerId = pyEnv["res.partner"].create({ name: "Bob" });
+    const channelId = pyEnv["discuss.channel"].create({
+        channel_member_ids: [
+            Command.create({ message_unread_counter: 0, partner_id: serverState.partnerId }),
+        ],
+        channel_type: "channel",
+        name: "General",
+    });
+    pyEnv["mail.message"].create({
+        body: "first message",
+        message_type: "comment",
+        model: "discuss.channel",
+        res_id: channelId,
+        author_id: bobPartnerId,
+    });
+    await start();
+    await openDiscuss(channelId);
+    await contains(".o-mail-Message");
+    await contains(".o-mail-Thread-newMessage hr + span", { count: 1, text: "New messages" });
+    await insertText(".o-mail-Composer-input", "hey!");
+    await click(".o-mail-Composer-send:enabled");
+    await contains(".o-mail-Message", { count: 2 });
+    await contains(".o-mail-Thread-newMessage hr + span", { count: 0, text: "New messages" });
+});
+
+test("remain when switching between chat window and discuss", async () => {
```

Comment:
Would clarify "[...] of same thread" _(just to removes a level of guess when reading test name by itself)_